src/Compilers/CSharp/Portable/Declarations/DeclarationTreeBuilder.cs

```diff
@@ -757,6 +758,17 @@ private SingleTypeDeclaration VisitTypeDeclaration(TypeDeclarationSyntax node, D
                     MessageID.IDS_FeaturePrimaryConstructors.CheckFeatureAvailability(diagnostics, node, node.SemicolonToken.GetLocation());
                 }
             }
+            else if (node.Kind() is SyntaxKind.ExtensionDeclaration)
+            {
+                if (node.ParameterList != null)
+                {
+                    diagnostics.Add(ErrorCode.ERR_UnexpectedParameterList, node.ParameterList.GetLocation());
+                }
+                else
+                {
+                    MessageID.IDS_FeatureExtensions.CheckFeatureAvailability(diagnostics, node, node.Keyword.GetLocation());
```

Comment:
>MessageID.IDS_FeatureExtensions.CheckFeatureAvailability(diagnostics, node, node.Keyword.GetLocation()); [](http://example.com/codeflow?start=20&length=104)

It feels like we should check for feature support regardless of the presence of `ParameterList`

Reply:
Was following the pattern above. Looks like we do the feature check only for trees that don't have specifically illegal Syntax.  

Personally, this seems fine to me.  If you have the PC, you are told it is illegal.  You remove it, and you find out you might need to upgrade your lang version.

Reply:
> Was following the pattern above

What pattern? For interfaces primary constructors are not supported regardless of language version. For classes and structs the version is checked unconditionally. There is no direct similarity with extensions, for which the check for parameter list and language version are orthogonal.

Reply:
For extensions, primary constructors are never allowed either. So that check happens unilaterally.

Reply:
> So that check happens unilaterally. 

An I am not suggesting to change that. The checks are unrelated, and, therefore, should be performed independently. Please change the code as suggested. To avoid any confusion this is the code I expect to see:
```
            else if (node.Kind() is SyntaxKind.ExtensionDeclaration)
            {
                if (node.ParameterList != null)
                {
                    diagnostics.Add(ErrorCode.ERR_UnexpectedParameterList, node.ParameterList.GetLocation());
                }

                MessageID.IDS_FeatureExtensions.CheckFeatureAvailability(diagnostics, node, node.Keyword.GetLocation());
            }
```

Reply:
wfm